### PR TITLE
document execution types

### PR DESCRIPTION
**Pull request by damip**

### CODE REVIEW

File: massa-execution/src/types.rs [hunk 1]
```diff
@@ -25,26 +26,49 @@ pub(crate) struct StepHistoryItem {
...
-/// Stateful context, providing an execution context to massa between module executions
+/// Stateful context, providing a context during the execution of a module
 pub(crate) struct ExecutionContext {
+    /// final and active ledger at the current step
     pub ledger_step: SCELedgerStep,
+
+    /// max gas for this execution
     pub max_gas: u64,
+
+    /// coins transferred at the launch of the execution
```
Comment (AureliaDolo) [inline massa-execution/src/types.rs:37]: To whom are the coins transferred ?

File: massa-execution/src/types.rs [hunk 2]
```diff
@@ -25,26 +26,49 @@ pub(crate) struct StepHistoryItem {
...
     pub call_stack: VecDeque<Address>,
+
+    /// list of addresses created so far during excution
     pub owned_addresses: Set<Address>,
+
+    /// True if it's a read-only context
     pub read_only: bool,
 }
 
+/// an active execution step request
```
Comment (AureliaDolo) [inline massa-execution/src/types.rs:65]: I'm confused by the "request", Isn't it something else ?

File: massa-execution/src/types.rs [hunk 3]
```diff
@@ -25,26 +26,49 @@ pub(crate) struct StepHistoryItem {
...
 #[derive(Clone)]
 pub(crate) struct ExecutionStep {
+    /// slot at which the execution step will happen
     pub slot: Slot,
-    // TODO add pos_seed for RNG seeding
-    // TODO add pos_draws to list the draws (block and endorsement creators) for that slot
+
+    /// Some(BlockID, block), if a block is present at this slot, otherwise None
     pub block: Option<(BlockId, Block)>, // None if miss
```
Comment (AureliaDolo) [inline massa-execution/src/types.rs:72]: The comment on this line is redundant

File: massa-execution/src/vm.rs [hunk 4]
```diff
@@ -16,10 +16,18 @@ use std::sync::{Arc, Mutex};
 use tokio::sync::oneshot;
 use tracing::debug;
 
+/// Virtual Machine and step history system
 pub(crate) struct VM {
+    /// thread count
```
Comment (AureliaDolo) [inline massa-execution/src/vm.rs:21]: Thread count is now in the static config, shouldn't it be available here, without the need to duplicate it here ? Or am I missing on something @yvan-sraka

Comment (yvan-sraka) [reply massa-execution/src/vm.rs:21]: You right!

Comment (AureliaDolo) [reply massa-execution/src/vm.rs:21]: And it can be removed from all functions ? Like Address::get_thread(thread_count) ?

Comment (damip) [reply massa-execution/src/vm.rs:21]: Can someone do this ? I'm curious to see how it's done in this particular case.  I am still confused about the new config system

File: massa-execution/src/types.rs (hunk 3 above)
Comment (damip) [reply massa-execution/src/types.rs]: removed

File: massa-execution/src/types.rs (hunk 2 above)
Comment (damip) [reply massa-execution/src/types.rs]: clarified

File: massa-execution/src/types.rs (hunk 1 above)
Comment (damip) [reply massa-execution/src/types.rs]: clarified

File: massa-execution/src/vm.rs (hunk 4 above)
Comment (AureliaDolo) [reply massa-execution/src/vm.rs:21]: see #2197

Comment (damip) [reply massa-execution/src/vm.rs:21]: ok then it will be done as a followup

Comment (adrien-zinger) [reply massa-execution/src/vm.rs:21]: I can make another PR this week if everybody is ok to resolve the #2197. For me we can merge this PR. @yvan-sraka @AureliaDolo @damip

**Review by AureliaDolo — APPROVED** [review]

File: massa-execution/src/types.rs [hunk 5]
```diff
@@ -26,26 +27,53 @@ pub(crate) struct StepHistoryItem {
...
+    /// block ID, if one is present at this slot
     pub opt_block_id: Option<BlockId>,
+
+    /// block creator addr, if there is a block at this slot
     pub opt_block_creator_addr: Option<Address>,
+
+    /// address call stack, most recent is at the back
     pub call_stack: VecDeque<Address>,
+
+    /// list of addresses created so far during excution, stacked
```
Comment (AureliaDolo) [inline massa-execution/src/types.rs:59]: Why is it a vec of vecs ?

File: massa-execution/src/types.rs [hunk 6]
```diff
@@ -35,8 +35,8 @@ pub(crate) struct ExecutionContext {
     /// max gas for this execution
     pub max_gas: u64,
 
-    /// coins transferred to the target address during a call
-    pub coins: Amount,
+    /// coins transferred to the target address during a call, stacked
```
Comment (AureliaDolo) [inline massa-execution/src/types.rs:38]: What do you mean by staked ?

File: massa-execution/src/types.rs (hunk 5 above)
Comment (damip) [reply massa-execution/src/types.rs:59]: because it's a vec<address> that is stacked

File: massa-execution/src/types.rs (hunk 6 above)
Comment (damip) [reply massa-execution/src/types.rs:38]: stacked, same way as the call stack

File: massa-execution/src/types.rs (hunk 5 above)
Comment (AureliaDolo) [reply massa-execution/src/types.rs:59]: I would add something like "stacked,in the same order as the call stack"

By the way, id module 1 calls module 2 calls module 3, and addresses are created only during module2 will this be the same length as the call stack ?

**Review by AureliaDolo — COMMENTED** [review]
That's probably an issue for another day, but how do you ensure that coin_stack and owned address stack are corresponding to the call stack ?

Maybe the call stack could be somthing like a VecDeque<(Address, coins: Amount, owned_addrs: Vec<Address>) Or am I missing something ?

File: massa-execution/src/types.rs (hunk 5 above)
Comment (damip) [reply massa-execution/src/types.rs:59]: > That's probably an issue for another day, but how do you ensure that coin_stack and owned address stack are corresponding to the call stack ?
> 
> Maybe the call stack could be somthing like a VecDeque<(Address, coins: Amount, owned_addrs: Vec
> ) Or am I missing something ?

Very good idea. Will refactor it as  `call_stack: Vec<StackElement>`:

```rust
struct StackElement {
    address: Address,
    coins: Amount,
    owned_addrs: Set<Address>
}
```

**Review by AureliaDolo — APPROVED** [review]

**Comment by damip** [comment]
bors merge

**Pull request merged**